dev/release/README.md

```diff
@@ -19,18 +19,121 @@
 
 # DataFusion Python Release Process
 
-This is a work-in-progress that will be updated as we work through the next release.
-
 ## Preparing a Release Candidate
 
-- Update the version number in Cargo.toml
-- Generate changelog
-- Tag the repo with an rc tag e.g. `0.7.0-rc1`
-- Create tarball and upload to ASF
-- Start the vote
+### Tag the Repository
+
+```bash
+git tag 0.7.0-rc1
+git push apache 0.7.0-rc1
+```
+
+### Publish Python Wheels to testpypi
+
+Pushing an rc tag to master will cause a GitHub Workflow to run that will build the Python wheels.
+
+Go to https://github.com/apache/arrow-datafusion-python/actions and look for an action named "Python Release Build"
+that has run against the pushed tag.
+
+Click on the action and scroll down to the bottom of the page titled "Artifacts". Download `dist.zip`.
+
+Upload the wheels to testpypi.
```

Comment:
I think it might be helpful for people with less experience to also include the twine config here (with redacting the credentials) since `--repository=testpypi` is a config-specific thing.